autode/hessians.py

```diff
@@ -312,3 +317,200 @@ def frequencies_proj(self) -> List[Frequency]:
         vib_freqs = self._eigenvalues_to_freqs(lambdas)
 
         return trans_rot_freqs + vib_freqs
+
+
+def calculate_numerical_hessian(species, *args, **kwargs) -> None:
+    """
+    Calculate a numerical Hessian by shifting atoms and evaluating the gradient
+    with or without central differences to evaluate the numerical derivatives.
+
+    ---------------------------------------------------------------------------
+    Arguments:
+        species: Species to calculate the numerical Hesssian for
+
+        *args: Arguments passed to the calculator
+
+        **kwargs: Keyword arguments passed to the calculator
+
+    Raises:
+        (ValueError): For unsupported input
+    """
+
+    calculator = _NumericalHessianCalculator(species, *args, **kwargs)
+    calculator.calculate()
+
+    species.hessian = calculator.hessian
+
+    return None
+
+
+class _NumericalHessianCalculator:
+
+    def __init__(self,
+                 species:   'autode.species.Species',
+                 method:    'autode.wrappers.base.Method',
+                 keywords:  'autode.wrappers.keywords.GradientKeywords',
+                 do_c_diff:  bool,
+                 num_delta: 'autode.values.Distance'
+                 ):
+
+        self._species = species
+        self._method = method
+        self._keywords = self._validated(keywords)
+
+        self._do_c_diff = do_c_diff
+        self._num_delta = num_delta
+
+        self._hessian = Hessian(np.zeros(shape=self._hessian_shape),
+                                units='Ha Å^-2')
+        self._calculated_rows = []
+
+        self._n_cores = max(Config.n_cores // self._n_rows, 1)
```

Comment:
Add comment on this is the number of cores per process, rather than the total utilised